.github/workflows/nightly.yml

```diff
@@ -0,0 +1,225 @@
+name: Garage CI Nightly
+
+env:
+  GARAGE_GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  COMMIT_RANGE: origin/${{ github.base_ref }}...origin/${{ github.head_ref }}
+  DOCKER_TAG: garage-ci-189158592
+  OWNER: "rlworkgroup"
+  DOCKER_CACHE_REPO: "garage-ci"
+  MJKEY_PATH: ${HOME}/.mujoco/mjkey.txt
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+
+jobs:
+  build_docker_container:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Login to GitHub Package Registry
+        run: echo ${{ secrets.CI_REGISTRY_TOKEN }} | docker login docker.pkg.github.com -u gitanshu --password-stdin
+      - name: Build Docker container
+        run: |
+          docker build . \
+            -f docker/Dockerfile \
+            --target garage-dev-18.04 \
+            -t "${DOCKER_TAG}" \
+            --build-arg GARAGE_GH_TOKEN \
+            --cache-from="docker.pkg.github.com/${OWNER}/${DOCKER_CACHE_REPO}/${DOCKER_TAG}"
+      - name: Push to cache (GitHub Package Registry)
+        run: |
+          docker tag "${DOCKER_TAG}" "docker.pkg.github.com/${OWNER}/${DOCKER_CACHE_REPO}/${DOCKER_TAG}"
+          docker push "docker.pkg.github.com/${OWNER}/${DOCKER_CACHE_REPO}/${DOCKER_TAG}"
+
+
+    check_pre_commit:
```

Comment:
is there any way to define  these jobs somewhere else and include them in both workflows? If they are copied, there is a high risk of updates to one but not the other happening.

Reply:
I think I might be able to merge the two workflows and set a conditional, but there is no way to template it as far as I know